docs/Installation.md

```diff
@@ -1,8 +1,134 @@
+
+```bash
+export E2E_PAC_GITHUB_APP_ID='<YOUR_APP_ID>'
+
+export E2E_PAC_GITHUB_APP_PRIVATE_KEY=$(base64 < /PATH/TO/YOUR/DOWNLOADED/PRIVATE_KEY.pem)
+```
+
+<p>
+Navigate back to <a href="https://github.com/settings/apps">your GitHub App</a>, select Install App and select your GitHub org (the one that you're using in `MY_GITHUB_ORG` env var). Feel free to install it to all repositories of that organization or the forked repositories currently used by <a href="(https://github.com/redhat-appstudio/e2e-tests/blob/main/tests/rhtap-demo/rhtap-demo.go)">rhtap-demo</a> and <a href="(https://github.com/redhat-appstudio/e2e-tests/blob/main/tests/build/build.go">build tests</a>
+</p>
+
+</details>
+
+For bootstrapping a cluster, run the following command:
+
+   ```bash
+      make local/cluster/prepare
+   ```
+
+More information about how to deploy RHTAP
+are in the [infra-deployments](https://github.com/redhat-appstudio/infra-deployments) repository.
+
+### Building and running the e2e tests
+
+Most of the tests could require you to have specific container image repo's created (if you're using your own container image org/user account (`QUAY_E2E_ORGANIZATION`) or your own GitHub organization (`MY_GITHUB_ORG`)
+In that case, before you run the test, make sure you have created
+* `test-images` repo in quay.io, i.e. `quay.io/<QUAY_E2E_ORGANIZATION>/test-images` and make it **public**
+  * also make sure that the docker config, that is encoded in the value of `QUAY_TOKEN` environment variable, contains a correct credentials required to push to `test-images` repo. And make sure the robot account or user account has the **write** permissions set for `test-images` repo which is required by the tests to push the generated artifacts.
+* fork following GitHub repositories to your org (specified in `MY_GITHUB_ORG` env var)
+  * https://github.com/redhat-appstudio-qe/devfile-sample-hello-world (for running build-service tests)
+  * https://github.com/redhat-appstudio-qe/hacbs-test-project (for rhtap-demo test)
+  * https://github.com/redhat-appstudio-qe/strategy-configs (for rhtap-demo test)
+
+You can use the following make target to build and run the tests:
+   ```bash
+      make local/test/e2e
+   ```
+
+Or build and run the tests without scripts:
+1. Install dependencies and build the tests:
+
+   ``` bash
+   # Install dependencies
+   $ go mod tidy
+   # Copy the dependencies to vendor folder
+   $ go mod vendor
+   # Create `e2e-appstudio` binary in bin folder. Please add the binary to the path or just execute `./bin/e2e-appstudio`
+   $ make build
+   ```
+
+2. Run the e2e tests:
+The `e2e-appstudio` command is the root command that executes all test functionality. To obtain all available flags for the binary please use `--help` flags. All ginkgo flags and go tests are available in `e2e-appstudio` binary.
+
+
+   ```bash
+      `./bin/e2e-appstudio`
+   ```
+
```

Comment:
```suggestion

Note: The binary must be updated by running `make build` every time there are new changes in the tests.
```

I've been asked a couple of times:
* "Hey do I need to push my changes somewhere in order for it to be picked up when I run the tests?",
* "Hey, why aren't my test changes working as expected?"
maybe we need to state this clearly somewhere if it hasn't been done already. wdyt?

Reply:
I don’t understand what you want to say by this

Reply:
When you locally update a test, you must run `make build` in order to update the binary.

This info needs to be put somewhere in the guide, as I suggested above.

Reply:
I don’t see this necessarily here. This pull request is about guidelines not about how to work binaries in general. This is more golang knowledges and should be treated in another issues

Reply:
> This pull request is about guidelines

Then tbh the scope of this PR has been confusing since the beginning.
For instance: why was [this huge amount of text](https://github.com/redhat-appstudio/e2e-tests/pull/784/files#diff-b335630551682c19a781afebcf4d07bf978fb1f8ac04c6bf87428ed5106870f5L40) moved from the README.md file to [Installation.md file](https://github.com/redhat-appstudio/e2e-tests/pull/784/files#diff-fcf6d384bb8308546cecce54bb9678c715f95c119b402f9b61a78e0cf3d8fbc4R8)?
None of that text is concerned with guidelines as per my understanding.

I think in that case this change should be reverted and handled as part of a separate ticket.

Reply:
Having so much in main Readme is not good. Main Readme should be an index about what we have in docs. Take a look to Kubernetes project

Reply:
> Having so much in main Readme is not good.

This is not a main README, it's an Installation.md file that _is_ supposed to explain things in detail, correct me if I'm wrong.

> Take a look to Kubernetes project

Kubernetes has its own documentation site where they can describe each step in great detail, but we don't. We can only rely on our files (within this repo) to make the whole process less vague because I've been answering some very repetitive questions for a long time.

Reply:
I still don’t see necessary to have an explanation about updating a binary files.

Reply:
Correct me if I'm wrong, but I thought the main motivation for improving the guide was to address feedback from team members and make the experience with working with e2e-tests as smooth as possible. And since Dheeraj's suggestion is based on real conversations with team members who were struggling with this problem, wouldn't it make sense to put that small note there?